tests/kafkatest/services/security/security_config.py

```diff
@@ -150,8 +152,16 @@ def __init__(self, context, security_protocol=None, interbroker_security_protoco
             'sasl.kerberos.service.name' : 'kafka'
         }
 
-    def client_config(self, template_props=""):
-        return SecurityConfig(self.context, self.security_protocol, client_sasl_mechanism=self.client_sasl_mechanism, template_props=template_props)
+    def client_config(self, template_props="", static_jaas_conf=True):
+        return SecurityConfig(self.context, self.security_protocol, client_sasl_mechanism=self.client_sasl_mechanism, template_props=template_props, static_jaas_conf=static_jaas_conf)
+
+    def client_config_setup(self, node, template_props=""):
```

Comment:
Maybe not critical, but can we rearrange this to use the existing single `client_config` method? Seems like we could potentially add just the `node` parameter, default `None`. I'm a bit unclear why our existing paths all use `static_jaas_conf=True` but the `client_config_setup` path uses `static_jaas_conf = self.has_sasl and self.has_ssl` -- since both `has_ssl` and `has_sasl` existed before, they don't seem like the right criteria (but I also haven't really paid attention to the relevant KIPs...). If there's a good reason for the difference, I think it should probably be clearer, either in naming, arguments triggering dynamic JAAS configs, or at least comments.